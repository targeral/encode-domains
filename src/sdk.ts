import path from 'path';
import { transformAsync } from '@babel/core';
import { fs } from '@modern-js/utils';

export const transformCodeAsync = async (code: string) => {
  const result = await transformAsync(code, {
    // eslint-disable-next-line @typescript-eslint/no-require-imports
    plugins: [require('./babel-plugin')],
  });

  if (result === null || !result.code) {
    throw new Error('Could not transform file');
  }

  return result.code;
};

export const transformFileAsync = async (
  targetFilePath: string,
  outputFilePath?: string,
) => {
  const content = await fs.readFile(targetFilePath, 'utf-8');
  const outputCode = await transformCodeAsync(content);

  const filePath =
    outputFilePath ?? path.join(process.cwd(), './encode-domain');

  fs.ensureFileSync(filePath);
  fs.writeFileSync(filePath, outputCode);
};

export const transformUrl = (str: string) => {
  return `atob(${btoa(str)})`;
};
