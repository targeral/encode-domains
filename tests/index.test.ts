import { transformCodeAsync } from '@/sdk';

describe('Default cases', () => {
  test(`const url = 'https://example.com'`, async () => {
    const code = `const url = 'https://example.com';`;
    const s = await transformCodeAsync(code);
    const expectCode = `const url = atob("aHR0cHM6Ly9leGFtcGxlLmNvbQ==")`;
    expect(s).toMatchSnapshot();
    expect(s).toContain(expectCode);
  });

  test(`const urlObj = {a: 'https://example.com'}`, async () => {
    const code = `const urlObj = {a: 'https://example.com'}`;
    const s = await transformCodeAsync(code);
    const expectCode = `a: atob("aHR0cHM6Ly9leGFtcGxlLmNvbQ==")`;
    expect(s).toMatchSnapshot();
    expect(s).toContain(expectCode);
  });

  test(`const urlObj = true ? 'https://example.com' : 'https://demo.com'`, async () => {
    const code = `const urlObj = true ? 'https://example.com' : 'https://demo.com';`;
    const s = await transformCodeAsync(code);
    const expectResult1 = `atob("aHR0cHM6Ly9leGFtcGxlLmNvbQ==")`;
    const expectResult2 = `atob("aHR0cHM6Ly9kZW1vLmNvbQ==")`;
    expect(s).toMatchSnapshot(s);
    expect(s).toContain(expectResult1);
    expect(s).toContain(expectResult2);
  });

  test(`const isSameUrl = 'https://example.com' === 'https://demo.com';`, async () => {
    const code = `const isSameUrl = 'https://example.com' === 'https://demo.com';`;
    const s = await transformCodeAsync(code);
    const expectResult1 = `atob("aHR0cHM6Ly9leGFtcGxlLmNvbQ==")`;
    const expectResult2 = `atob("aHR0cHM6Ly9kZW1vLmNvbQ==")`;
    expect(s).toMatchSnapshot(s);
    expect(s).toContain(expectResult1);
    expect(s).toContain(expectResult2);
  });

  test(`const urlObj = {a: '//example.com'} with urls option`, async () => {
    const code = `const urlObj = {a: '//example.com'}`;
    const s = await transformCodeAsync(code, { urls: ['//example.com'] });
    expect(s).toMatchSnapshot();
  });

  // TODO: webpack test
});
