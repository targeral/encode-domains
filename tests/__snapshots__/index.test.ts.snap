// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default cases const isSameUrl = 'https://example.com' === 'https://demo.com';: const isSameUrl = atob("aHR0cHM6Ly9leGFtcGxlLmNvbQ==") === atob("aHR0cHM6Ly9kZW1vLmNvbQ=="); 1`] = `"const isSameUrl = atob("aHR0cHM6Ly9leGFtcGxlLmNvbQ==") === atob("aHR0cHM6Ly9kZW1vLmNvbQ==");"`;

exports[`Default cases const url = 'https://example.com' 1`] = `"const url = atob("aHR0cHM6Ly9leGFtcGxlLmNvbQ==");"`;

exports[`Default cases const urlObj = {a: '//example.com'} with urls option 1`] = `
"const urlObj = {
  a: atob("Ly9leGFtcGxlLmNvbQ==")
};"
`;

exports[`Default cases const urlObj = {a: 'https://example.com'} 1`] = `
"const urlObj = {
  a: atob("aHR0cHM6Ly9leGFtcGxlLmNvbQ==")
};"
`;

exports[`Default cases const urlObj = true ? 'https://example.com' : 'https://demo.com': const urlObj = true ? atob("aHR0cHM6Ly9leGFtcGxlLmNvbQ==") : atob("aHR0cHM6Ly9kZW1vLmNvbQ=="); 1`] = `"const urlObj = true ? atob("aHR0cHM6Ly9leGFtcGxlLmNvbQ==") : atob("aHR0cHM6Ly9kZW1vLmNvbQ==");"`;
